src/libs/memoize/cache/arrayCacheBuilder.ts

```diff
@@ -0,0 +1,76 @@
+import type {Cache, CacheOpts} from '@libs/memoize/types';
+import {getEqualityComparator} from '@libs/memoize/utils';
+
+/**
+ * Builder of the cache using `Array` primitive under the hood.
+ * @param opts - Cache options, check `CacheOpts` type for more details.
+ * @returns
+ */
+function buildArrayCache<K extends unknown[], V>(opts: CacheOpts): Cache<K, V> {
+    const cache: Array<[K, V]> = [];
+
+    const keyComparator = getEqualityComparator(opts);
+
+    function getKeyIndex(key: K) {
+        for (let i = cache.length - 1; i >= 0; i--) {
```

Comment:
Let's add a comment:

```
// We search the array backwards because the most recently added entries are at the end, and our heuristic follows the principles of an LRU cache - that the most recently added entries are most likely to be used again.
```